packages/experimental/navigation/otel/hooks/useTrace.ts

```diff
@@ -0,0 +1,39 @@
+import {trace, Tracer, TracerProvider} from '@opentelemetry/api';
+import {MutableRefObject, useEffect, useRef} from 'react';
+
+const TRACER_DEFAULT = {
+  name: 'default',
+  version: '1.0',
+};
+
+interface ConfigArgs {
+  name?: string;
+  version?: string;
+}
+
+type TracerRef = MutableRefObject<Tracer | null>;
+
+const useTrace = (
+  {name, version}: ConfigArgs = {},
+  provider: TracerProvider,
+): TracerRef => {
+  // otel
+  const tracerRef = useRef<Tracer | null>(null);
+
+  // using the layout effect to make sure the tracer is initialized before the component is rendered
+  useEffect(() => {
+    if (tracerRef.current === null && provider) {
+      trace.setGlobalTracerProvider(provider);
```

Comment:
I think we should avoid this in the library since it may have an unintended effect on the rest of the app's instrumentation, should be able to do `provider.getTracer(...)` without setting it to be global